src/test/java/hudson/plugins/batch_task/BatchTaskTest.java

```diff
@@ -79,10 +89,11 @@ public void testExecute() throws Exception {
                     "echo \"$TASK_ID:$GLOBAL:$OVERRIDE_ME:$HUDSON_USER\"\n");
         }
         p.addProperty(new BatchTaskProperty(task));
-        p.scheduleBuild2(0).get();
-        new WebClient().getPage(p, "batchTasks/task/test/execute");
-        Queue.Item q = hudson.getQueue().getItem(task);
-        if (q!=null) q.getFuture().get(5, TimeUnit.SECONDS);
+        p.scheduleBuild2(0);
+        for (int i=0; i<10 && p.getLastBuild()==null; i++) {
```

Comment:
Just a suggestion. Instead of checking the lastBuild every 100 seconds 10 times, you can use the `FreeStyleBuild.isBuilding()` method in a `while` statement.

```
FreeStyleBuild b = p.scheduleBuild2(0).get();
while (b.isBuilding()) {
    Thread.sleep(100);
}
```

I think is more robust. Maybe the job does not finish in 10*100 ms